app/routes/offer/offer_details.js

```diff
@@ -1,19 +1,28 @@
-import AuthorizeRoute from './../authorize';
+import AuthorizeRoute from "./../authorize";
 
 export default AuthorizeRoute.extend({
-
   model() {
-    if(this.modelFor('offer')){
-      var offerId = this.modelFor('offer').get('id');
-      return this.store.peekRecord('offer', offerId);
+    if (this.modelFor("offer")) {
+      var offerId = this.modelFor("offer").get("id");
+      return this.store.peekRecord("offer", offerId);
     }
   },
 
   afterModel(my_offer) {
-    if(!my_offer) {
-      this.transitionTo('offers');
-    } else if(my_offer && my_offer.get('itemCount') === 0) {
-      this.transitionTo('offer', my_offer);
+    this.store.query("message", {
+      messageable_type: "Offer",
+      messageable_id: my_offer.get("id")
+    });
+
+    this.store.query("message", {
+      messageable_type: "Item",
+      messageable_id: my_offer.get("items").getEach("id")
+    });
```

Comment:
unless there's a very specific reason you're not doing it, you should await promises. Otherwise it's a recipe for race-conditions

Reply:
async-await does not work with ember 2.11 version. we can fix this in ember upgrade ticket.

Reply:
@swatijadhav if async/await does not work, it's not a reason not to await a promise. You can use `Promise.then` in that case

Reply:
```suggestion
return Ember.RSVP.Promise.all([
    this.store.query("message", { messageable_type: "Offer", messageable_id: my_offer.get("id")}), 
    this.store.query("message", { messageable_type: "Item", messageable_id: my_offer.get("items").getEach("id")})
]);
```